src/lib/glifApi.ts

```diff
@@ -94,3 +94,61 @@ export const getAllowanceForClient = async (
     }
   }
 }
+
+/**
+ * Wait for a message to appear on chain and get it.
+ *
+ * @param {string} cid - Transaction CID.
+ * @returns {Promise<ApiAStateWaitMsgResponse>} ApiAStateWaitMsgResponse - The response from the API.
+ */
+export const getStateWaitMsg = async (
+  cid: string,
+): Promise<ApiAStateWaitMsgResponse> => {
+  try {
+    const requestBody = {
+      jsonrpc: '2.0',
+      method: 'Filecoin.StateWaitMsg',
+      params: [
+        {
+          '/': cid,
+        },
+        1,
+        10,
+        true,
+      ],
+      id: 1,
+    }
+    const response = await fetch(`${config.glifNodeUrl}`, {
+      method: 'POST',
+      body: JSON.stringify(requestBody),
+    })
+
+    const data = await response.json()
+    if (data?.error && data.error.code === -32603) {
+      return await getStateWaitMsg(cid)
+    }
+
+    if (data?.error) {
+      return {
+        data: '',
+        error: data.error.message,
+        success: false,
+      }
+    }
+
+    return {
+      data: data.result,
+      error: '',
+      success: true,
```

Comment:
Based on previous usage of `data?` I assume data can be null/undefined. Won't it fail here if we try to take `result` from null/undefined?
And we probably shouldn't say success: true if we got no data?